apps/heft/src/cli/HeftToolsCommandLineParser.ts

```diff
@@ -189,13 +191,5 @@ export class HeftToolsCommandLineParser extends CommandLineParser {
       this._terminal.writeLine();
       this._terminal.writeErrorLine(error.stack!);
     }
-
-    await this._metricsCollector.flushAndTeardownAsync();
-
-    if (!process.exitCode || process.exitCode > 0) {
-      process.exit(process.exitCode);
```

Comment:
How does this not reintroduce the issue where a failed build returns a zero exit code?

Reply:
Fair enough. I can revert that commit.